#!/usr/bin/Rscript

#This script combines output from summary statistics generated by Samtools and Bedtools
#It requires input in tab-separated format 


#Import libraries
library(methods)
library(plyr)
library(dplyr)
library(magrittr)
library(reshape2)

#Set up argument parser and QC-------------------------------------------------
args <- commandArgs()

#print(args)
cov_fp <- args[6]
read_fp <-args[7]
output <-args[8]

if(is.na(cov_fp)) 
{
  stop("Coverage filepath not defined.")
}

if(is.na(read_fp)) 
{
  stop("Mapped read filepath not defined.")
}

if(is.na(output)) 
{
  stop("Output file not defined")
}


##Build a dataframe of coverage after very sensitive local alignment of paired reads. Works.

fc_files <- list.files(path=cov_fp, 
                       recursive = TRUE, 
                       full.names = TRUE, 
                       pattern = glob2rx("*_genomecoverage.txt"))

fc_df <- data.frame(Virus=factor(), 
                    Coverage=numeric(),
                    AlignmentSample=factor(), 
                    SampleID=factor(),
                    StudyID=factor())

for (i in 1:length(fc_files)) {
  file_name = fc_files[i]
  if(file.info(fc_files[i])$size != 0) { 
    alignment_name = sub("_genomecoverage.txt", "", file_name) %>% 
      sub(".*Coverage/", "", .) 
    
    study_name = sub("_genomecoverage.txt", "", file_name) %>% 
      sub(".*Alignments/", "", .) %>% 
      sub("/Coverage.*", "", .)
    
    sample <- read.delim(file_name, sep="\t", header=F) 
    sample$V3 <- alignment_name
    sample$V4 <- sub("all_viruses_", "", sample$V3)
    sample$V5 <- study_name
    colnames(sample) <- c("Virus", "Coverage", "AlignmentSample", 
                          "SampleID", "StudyID")
    fc_df <- rbind(fc_df,sample) 
  }
}

fc_df <- mutate(fc_df, Virus = droplevels(Virus), 
                AlignmentSample = as.factor(AlignmentSample), 
                SampleID = as.factor(SampleID), 
                StudyID = as.factor(StudyID))

#Clean up
rm(file_name)
rm(fc_files)
rm(i)
rm(study_name)
rm(alignment_name)
rm(sample)