#-*- mode: Snakemake -*-

rule align_reads:
	input:
		r1 = str(DATA_DIR+"/{sample}_1.fastq"),
		r2 = str(DATA_DIR+"/{sample}_2.fastq"),
		index = str(TARGETS+".1.bt2")
	output:
		temp(str(OUTPUT_DIR+"/alignments/{sample}.sam"))
	params:
		index = str(TARGETS)
	shell:
		"bowtie2 -q --local -t --very-sensitive-local --no-mixed --no-discordant -x {params.index} -1 {input.r1} -2 {input.r2} -S {output}"
<<<<<<< HEAD

rule process_alignment:
	input:
		str(OUTPUT_DIR+"/alignments/{sample}.sam")
	output:
		bam = temp(str(OUTPUT_DIR+"/alignments/{sample}.bam")),
		sorted = str(OUTPUT_DIR+"/alignments/{sample}.sorted.bam"),
		bai = str(OUTPUT_DIR+"/alignments/{sample}.sorted.bam.bai"),
	params:
		target = str(TARGETS)
	shell:
		"""
		samtools view -bT {params.target} {input} > {output.bam}
		samtools sort -o {output.sorted} {output.bam}
		samtools index {output.sorted} {output.bai}
		"""

rule all_sample_summary:
	input:
		expand(OUTPUT_DIR+"/alignments/{sample}.sorted.bam",sample=config["samples"])
	output:
		str(OUTPUT_DIR+"/summary/temp_sample_summary.txt")
	params:
		sum_dir = str(OUTPUT_DIR + "/summary/"),
		aln_dir = str(OUTPUT_DIR + "/alignments/")
	shell:
		"ls {params.aln_dir}*sorted.bam | wc -l > {params.sum_dir}temp_sample_summary.txt"



=======
>>>>>>> 501c78e425b10b1bb4e7e0bc172504e7b7fc21dd
