# -*- mode: Snakemake -*-

rule mapping_stats:
	input: 
		bam = str(OUTPUT_DIR+"/alignments/{ref}/{sample}.sorted.bam"),
		idx = str(OUTPUT_DIR+"/alignments/{ref}/{sample}.sorted.bam.bai")
	output:
		temp(str(OUTPUT_DIR+"/alignments/{ref}/{sample}.sorted.idxstats.tsv"))
	shell:
		"""
		samtools idxstats {input.bam} > {output}
		"""
rule calculate_coverage:
	input:
		bam = str(OUTPUT_DIR+"/alignments/{ref}/{sample}.sorted.bam"),
		idx = str(OUTPUT_DIR+"/alignments/{ref}/{sample}.sorted.bam.bai")
	output:
		str(OUTPUT_DIR+"/alignments/{ref}/{sample}.genomecoverage.txt")
	shell:
		"""
		samtools view -b {input.bam}|genomeCoverageBed -ibam stdin|grep -v 'genome'| perl scripts/coverage_counter.pl > {output}	
		"""
rule combine_coverage_stats:
	input:
		cov = str(OUTPUT_DIR+"/alignments/{ref}/{sample}.genomecoverage.txt"),
		stats = str(OUTPUT_DIR+"/alignments/{ref}/{sample}.sorted.idxstats.tsv")
	output:
		str(OUTPUT_DIR+"/alignments/{ref}/{sample}.align.summary.txt")

	shell:
		"""
		Rscript scripts/summarize_alignments.R {input.cov} {input.stats} {output}
		"""

#Combine information for all samples into a single file
rule all_summary:
	input:
		expand(OUTPUT_DIR+"/alignments/{ref}/{sample}.align.summary.txt",sample=config["samples"], ref = ref_names)
	output:
		str(OUTPUT_DIR+"/summary/all_align_summary.txt")
	params:
		align_dir = str(OUTPUT_DIR + "/alignments/")
	shell:
		"""
		echo -e "Sample\tAlignTarget\tFractionCoverage\tTargetLength\tMappedReads" > {output}
		cat {params.align_dir}*.align.summary.txt >> {output}
		"""

